bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -212,7 +212,7 @@ public ObjectId(final String hexString) {
      * @throws IllegalArgumentException if array is null or not of length 12
      */
     public ObjectId(final byte[] bytes) {
-        this(ByteBuffer.wrap(notNull("bytes", bytes)));
+        this(ByteBuffer.wrap(isTrueArgument("bytes has length of 12", bytes, notNull("bytes", bytes).length == 12)));
```

Comment:
Using it in concert with notNull is a bit of a mouthful, but at least the exception message will be correct for both failure conditions